server/src/main/java/com/google/musicanalysis/cache/AnalysisCache.java

```diff
@@ -1,87 +1,115 @@
 package com.google.musicanalysis.cache;
 
 import com.google.musicanalysis.types.*;
-import java.lang.Object;
-import java.lang.ClassNotFoundException;
-import java.time.Instant;
-import java.net.URI;
-import java.util.List;
-import java.util.Map;
-import java.util.HashMap;
+import java.io.BufferedInputStream;
+import java.io.BufferedOutputStream;
+import java.io.EOFException;
 import java.io.File;
-import java.io.InputStream;
 import java.io.FileInputStream;
-import java.io.ObjectInputStream;
 import java.io.FileOutputStream;
-import java.io.ObjectOutputStream;
 import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.util.HashMap;
+import javax.crypto.BadPaddingException;
+import javax.crypto.Cipher;
+import javax.crypto.CipherInputStream;
+import javax.crypto.CipherOutputStream;
+import javax.crypto.IllegalBlockSizeException;
+import javax.crypto.NoSuchPaddingException;
+import javax.crypto.SealedObject;
+import javax.crypto.spec.SecretKeySpec;
 
+enum FileStatus {
+    SUCCESS,
+    CREATED,
+    EXISTS
+}
 
 public class AnalysisCache {
-    private HashMap<String, AnalysisGroup> responseMap;
-    
-    public AnalysisCache(){
-        this.responseMap = new HashMap<String, AnalysisGroup>();
-    }
 
-    public void open() {
-        HashMap<String, AnalysisGroup> responseData = null;
+  private HashMap<String, AnalysisGroup> responseMap;
+  Cipher cipher;
 
-        int file = createFile(); // this will have use later.
+  public AnalysisCache() throws NoSuchAlgorithmException, NoSuchPaddingException {
+    this.responseMap = new HashMap<String, AnalysisGroup>();
+    this.cipher = Cipher.getInstance("Blowfish");
+  }
 
-        try {
-            // .ser for serialized 
-            FileInputStream inFile = new FileInputStream("cachedData.txt");
-            ObjectInputStream inData = new ObjectInputStream(inFile);
-            responseMap = (HashMap<String, AnalysisGroup>)inData.readObject();
-            inData.close();
-            inFile.close();
-        } catch (IOException e) {
-            e.printStackTrace();
-            return;
-        } catch (ClassNotFoundException c) {
-            System.out.println("Analysis Group class not found");
-            c.printStackTrace();
-            return;
-        }
-    }
+  public void open() throws InvalidKeyException, IllegalBlockSizeException, BadPaddingException {
```

Comment:
Instead of needing to open this file every time someone makes a web request, can we keep that cache persistent between calls? Maybe only periodically write to the file? File reads/writes are very expensive compared to reads/writes from memory.